rscripts/YAL5-single-codons.R

```diff
@@ -0,0 +1,1108 @@
+# the feature of interest it will be extracted by the following if statement
+
+if (file.exists(feature_of_interest)) {
+  feature_of_interest <- read.csv(feature_of_interest)
+  feature_of_interest <- feature_of_interest[, 1]
+}
+
+# read in GFF
+gff_df <- readGFFAsDf(gff)
+
+## TEST: When running on tidy sim data set, the gff appears as follows
+# There are 2 genes, MAT and MIKE, which have CDSs flanked by 20nt
+
+# > gff_df
+# # A tibble: 6 x 10
+# seqnames start   end width strand source   type  score phase Name
+# <fct>    <int> <int> <int> <fct>  <fct>    <fct> <dbl> <int> <chr>
+#   1 MAT          1    20    20 +      ewallace UTR5     NA    NA MAT
+# 2 MAT         21    32    12 +      ewallace CDS      NA    NA MAT
+# 3 MAT         33    52    20 +      ewallace UTR3     NA    NA MAT
+# 4 MIKE         1    20    20 +      ewallace UTR5     NA    NA MIKE
+# 5 MIKE        21    35    15 +      ewallace CDS      NA    NA MIKE
+# 6 MIKE        36    55    20 +      ewallace UTR3     NA    NA MIKE
+
+# Get gene_names
+gene_names <- unique(gff_df$Name)
+
+# Import the .tsv file of codon positions:
+yeast_codon_pos_i200 <- suppressMessages(readr::read_tsv(file = yeast_codon_table))
+
+# > str(yeast_codon_pos_i200)
+# spec_tbl_df [2,826,757 x 3] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
+# $ Gene    : chr [1:2826757] "YAL068C" "YAL068C" "YAL068C" "YAL068C" ...
+# $ PosCodon: num [1:2826757] 1 2 3 4 5 6 7 8 9 10 ...
+# $ Codon   : chr [1:2826757] "ATG" "GTC" "AAA" "TTA" ...
+
+
+### Functions to map number of reads to codon positions ###
+
+#' GetGeneCodonPosReads1dsnap(): uses snapgene to assign reads to codons.
+#' @param gene The gene to get datamatrix and reads for.
+#' @param dataset The name of dataset stored in .h5 file.
+#' @param hd_file The path to the .h5 hdf5 file holding read data for all genes,
+#' created from BAM files for dataset samples.
+#' @param min_read_length An integer, the minimum read length in H5 output;
+#' Default = 10 (set in generate_stats_figs.R from yaml)
+#' @param snapdisp An integer for additional displacement in the snapping
+#'
+#' @return A numeric object containing read numbers. each number represents
+#' reads at the next codon in the gene
+
+GetGeneCodonPosReads1dsnap <- function(gene, dataset, hd_file,
+                                       min_read_length, asite_displacement_length,
+                                       left, right,
+                                       snapdisp) {
+  # Get the matrix of read counts
+  reads_pos_length <- GetGeneDatamatrix(gene, dataset, hd_file)
+
+  # Assign reads to their A site nt, determined by read length
+  reads_asitepos <- CalcAsiteFixed(reads_pos_length,  
+                                   min_read_length,
+                                   asite_displacement_length
+  )
+  # Assign reads from nt positions to codon positions
+  SnapToCodon(reads_asitepos, left, right, snapdisp)
+}
+
+## TEST:: CalAsiteFixed:
```

Comment:
Are tests for CalcAsiteFixed() necessary as it is sourced from `read_count_functions.R`?

Reply:
tests removed